packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -218,6 +218,28 @@ export class TaskDefinition extends TaskDefinitionBase {
    */
   public defaultContainer?: ContainerDefinition;
 
+  /**
+   * The number of cpu units used by the task.
+   *
+   * If you are using the EC2 launch type, this field is optional and any value can be used.
+   * If you are using the Fargate launch type, this field is required and must be set,
+   */
+  public cpu?: string;
+
+  /**
+   * The amount (in MiB) of memory used by the task.
+   *
+   * If you are using the EC2 launch type, this field is optional and any value can be used.
+   * If you are using the Fargate launch type, this field is required and must be set,
+   */
+  public memoryMiB?: string;
```

Comment:
You are documenting this as if the user is supplying it, but this is a property of the class, so the user can only be reading it here.

Reply:
`memoryMiB` was a number on purpose (for usability), and so the suffix `MiB` makes sense because it adds a unit to the number. What if `memoryMiB` contains the string `"4G"`?

Reply:
Internally we should convert anything with a suffix to MiB. That being said, there was some discussion a few weeks ago about standardizing Memory (similar to what we do for Duration): https://github.com/aws/aws-cdk/issues/5088